src/me/mrletsplay/shittyauthlauncher/ShittyAuthSettingsController.java

```diff
@@ -32,7 +137,138 @@ void buttonSave(ActionEvent event) {
 	void buttonCancel(ActionEvent event) {
 		ShittyAuthLauncher.settingsStage.hide();
 	}
-	
+
+	@FXML
+	void buttonNewMirror(ActionEvent event) {
+		DownloadsMirror mirror = showEditMirrorDialog(null);
+		if(mirror != null) {
+			mirror.custom = true;
+			mirrors.add(mirror);
+			JSONArray checkedMirrors = new JSONArray();
+			for(DownloadsMirror loopMirror: mirrors){
+				if(loopMirror.custom){
+					System.out.println(loopMirror.name);
+					JSONObject mirrorJSON = new JSONObject();
+					mirrorJSON.set("name", loopMirror.name);
+					mirrorJSON.set("versionManifest", loopMirror.versionManifest);
+					mirrorJSON.set("assetsURL", loopMirror.assetsURL);
+					mirrorJSON.set("custom", loopMirror.custom);
+					checkedMirrors.add(mirrorJSON);
```

Comment:
Same when saving, just do
```java
checkedMirrors.add(loopMirror.toJSON(SerializationOption.DONT_INCLUDE_CLASS));
```
(Including the class doesn't hurt but isn't necessary in this case, that's why I'm passing `SerializationOption.DONT_INCLUDE_CLASS`)

Also, the sysout is probably not necessary.